app/src/main/java/nl/tudelft/bbw/controller/BlockController.java

```diff
@@ -261,4 +279,39 @@ private boolean verifySignature(byte[] signature, byte[] message) {
         }
     }
 
+    /**
+     * Method to add the multichain (the pairing person database) into your database
+     *
+     * @param multichain given the multichain of the pairing partner
+     * @throws BlockAlreadyExistsException
+     * @throws HashException
+     */
+    public void addMultichain(List<List<Block>> multichain) throws BlockAlreadyExistsException, HashException {
+        if (multichain.isEmpty()) {
+            return;
+        }
+        for (List<Block> chain : multichain) {
+            for (Block block : chain) {
+                if (block.getBlockType() == BlockType.GENESIS) {
```

Comment:
What happens if there is already a block in the database, and you want to add that same block to your chain. Currently, it will throw a runtime exception, while this should not be the case.